src/metrics/index.js

```diff
@@ -99,8 +104,50 @@ const MetricsFactory = context => {
       .then(() => latencyTrackerStop());
   };
 
+  const pushImpressionsCount = () => {
+    const imprCounts = impressionsCounter.size();
+    if (!impressionsCounter || imprCounts === 0) return Promise.resolve();
+
+    const pf = [];
+    const impressionsCount = impressionsCounter.popAll();
+    log.info(`Pushing ${imprCounts} impressions count`);
+
+    const keys = Object.keys(impressionsCount);
+    for (let i = 0; i < keys.length; i++) {
+      const splitted = keys[i].split('::');
+      if (splitted.length !== 2) continue;
+      const featureName = splitted[0];
+      const timeFrame = splitted[1];
+
+      const impressionsInTimeframe = {
+        t: featureName, // Test Name
+        m: timeFrame, // Time Frame
+        rc: impressionsCount[keys[i]] // Count
+      };
+
+      pf.push(impressionsInTimeframe);
+    }
+
+    return impressionsService(impressionsCountRequest(settings, {
+      body: JSON.stringify({ pf })
+    }))
+      .then(() => {
+        impressionsRetries = 0;
+      })
+      .catch(err => {
+        if (impressionsRetries) { // For now we retry only once.
+          log.warn(`Droping ${imprCounts} impressions count after retry. Reason ${err}.`);
```

Comment:
`Droping count of impressions for ${imprCounts} features after retry. Reason ${err}.`